File: app/src/main/java/info/nightscout/androidaps/interfaces/Interval.java
package info.nightscout.androidaps.interfaces;

/**
 * Created by mike on 21.05.2017.
 */

public interface Interval {
    long durationInMsec();
    long start();

    // planned end time at time of creation
    long originalEnd();

    // end time after cut
    long end();

    void cutEndTo(long end);
    boolean match(long time);
    boolean before(long time);
    boolean after(long time);

    boolean isInProgress();
    boolean isEndingEvent();

    boolean isValid();
}

File: core/src/main/java/info/nightscout/androidaps/interfaces/Interval.kt
package info.nightscout.androidaps.interfaces

interface Interval {

    fun durationInMsec(): Long
    fun start(): Long

    // planned end time at time of creation
    fun originalEnd(): Long

    // end time after cut
    fun end(): Long
    fun cutEndTo(end: Long)
    fun match(time: Long): Boolean
    fun before(time: Long): Boolean
    fun after(time: Long): Boolean
    val isInProgress: Boolean
    val isEndingEvent: Boolean
    val isValid: Boolean
}